### PR TITLE
NEW: In-memory cache for relation differ.

### PR DESCRIPTION
**Pull request by mfendeksilverstripe**
# NEW: In-memory cache for relation differ

* In-memory cache for relation differ
* finishing incomplete implementation of in-memory cache
* unit tests cover both cache hits & misses
* minor code cleanup

### CODE REVIEW

File: src/RelationDiffer/ClearCacheExtension.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace SilverStripe\Snapshots\RelationDiffer;
+
+use SilverStripe\Core\Extension;
+use SilverStripe\ORM\DataObject;
+
+/**
+ * Clear in-memory cache when DB update is executed
+ *
+ * @method DataObject|$this getOwner()
```
Comment (blueo) [inline src/RelationDiffer/ClearCacheExtension.php:11]: maybe not needed

**Review by blueo — COMMENTED** [review]
Implementation looks fine - I'm just trusting that there is a need for this memory cache as its a bit of overhead if it isn't necessary.

File: src/RelationDiffer/ClearCacheExtension.php (hunk 1 above)
Comment (mfendeksilverstripe) [reply src/RelationDiffer/ClearCacheExtension.php:11]: Not used currently but keeping it there for the sake of docblock consistency.

**Comment by mfendeksilverstripe** [comment]
@blueo Relation diff is quite expensive operation and this is why the in-memory cache was part of the initial codebase during the handover, although not fully implemented. Local test runs show that there is benefit in having this cache.

**Review by blueo — APPROVED** [review]

**Pull request merged**